zyte_spider_templates_project/settings.py

```diff
@@ -17,10 +17,15 @@
     "duplicate_url_discarder.Addon": 600,
     "zyte_spider_templates.Addon": 700,
 }
-
+DOWNLOADER_MIDDLEWARES = {
+    "scrapy_poet.InjectionMiddleware": 543,
+    "scrapy.downloadermiddlewares.stats.DownloaderStats": None,
+    "scrapy_poet.DownloaderStatsMiddleware": 850,
+}
 SPIDER_MIDDLEWARES = {
     "scrapy_poet.RetryMiddleware": 275,
 }
+REQUEST_FINGERPRINTER_CLASS = "scrapy_poet.ScrapyPoetRequestFingerprinter"
```

Comment:
This interacts both with the duplicate url discarder and with scrapy-zyte-api (see https://github.com/zytedata/duplicate-url-discarder/blob/d6920865039db10d1eb0a6763d1fa3be20e3469b/duplicate_url_discarder/_addon.py#L26 and https://github.com/scrapy-plugins/scrapy-zyte-api/blob/cc3cfddcd5e38502d7cda211178aafcd0a8863d5/scrapy_zyte_api/addon.py#L76). 

I don't expect any issues, but we should check somehow that it works.

Reply:
Working on a test for it, I found out it was working as expected before this change. It’s because [the scrapy-zyte-api fingerprinter already uses the scrapy-poet one as fallback if scrapy-poet is installed](https://github.com/scrapy-plugins/scrapy-zyte-api/blob/cc3cfddcd5e38502d7cda211178aafcd0a8863d5/scrapy_zyte_api/_request_fingerprinter.py#L33-L40). So this line is unnecessary.

Reply:
On a related note, the scrapy-zyte-api add-on also [configures the injection middleware](https://github.com/scrapy-plugins/scrapy-zyte-api/blob/cc3cfddcd5e38502d7cda211178aafcd0a8863d5/scrapy_zyte_api/addon.py#L118),